website/docs/r/vapp_firewall_rules.html.markdown

```diff
@@ -0,0 +1,119 @@
+    destination_port = "any"
+    destination_ip   = "any"
+    source_port      = "any"
+    source_ip        = "10.10.0.0/24"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations.
+* `vdc` - The name of VDC to use, optional if defined at provider level.
+* `vapp_id` - (Required) The identifier of [vApp](/docs/providers/vcd/r/vapp.html).
+* `network_id` - (Required) The identifier of [vApp network](/docs/providers/vcd/r/vapp_network.html).
+* `default_action` - (Required) Either 'allow' or 'drop'. Specifies what to do should none of the rules match.
+* `log_default_action` - (Optional) Flag to enable logging for default action. Default value is `false`.
+* `rule` - (Optional) Configures a firewall rule; see [Rules](#rules) below for details.
+
+<a id="rules"></a>
+## Rules
+
+Each firewall rule supports the following attributes:
+
+* `name` - (Optional) Name of the firewall rule.
+* `enabled` - (Optional) `true` value will enable firewall rule. Default is `true`.
+* `policy` - (Optional) Specifies what to do when this rule is matched. Either `allow` or `drop`.
+* `protocol` - (Optional) The protocol to match. One of `tcp`, `udp`, `icmp`, `any` or `tcp&udp`. Default is `any`.
+* `destination_port` - (Optional) The destination port to match. Either a port number or `any`.
+* `destination_ip` - (Optional) The destination IP to match. Either an IP address, IP range or `any`.
+* `destination_vm_id` - (Optional) Destination VM identifier.
+* `destination_vm_ip_type` - (Optional) The value can be one of: `assigned` - use assigned internal IP, `NAT` - use NATed external IP.
+* `destination_vm_nic_id` - (Optional) VM NIC ID to which this rule applies.
+* `source_port` - (Optional) The source port to match. Either a port number or `any`.
+* `source_ip` - (Optional) The source IP to match. Either an IP address, IP range or `any`.
+* `source_vm_id` - (Optional) Source VM identifier.
+* `source_vm_ip_type` - (Optional) The value can be one of: `assigned` - use assigned internal IP, `NAT` - use NATed external IP.
+* `source_vm_nic_id` - (Optional) VM NIC ID to which this rule applies.
+* `enable_logging`- (Optional) `true` value will enable rule logging. Default is `false`.
+
+## Importing
+
+~> **Note:** The current implementation of Terraform import can only import resources into the state.
+It does not generate configuration. [More information.](https://www.terraform.io/docs/import/)
+
+An existing an vApp network firewall rules can be [imported][docs-import] into this resource
+via supplying the full dot separated path to vapp network. An example is
+below:
+
+```
+terraform import vcd_vapp_firewall_rules.my-rules my-org.my-vdc.vapp_name.network_name
+```
+or using IDs:
+```
+terraform import vcd_vapp_firewall_rules.my-rules my-org.my-vdc.vapp_id.network_id
```

Comment:
How the end user can get the vApp network ID? If no easy way, what about providing the list function like we had for other resources?
CC: @dataclouder

Reply:
+1 
I missed this in my review.

Reply:
added list functionality

Reply:
This is great, thanks!